project-koku/koku
# [COST-3953] Masu resummarize by provider_type

## Jira Ticket

[COST-3953](https://issues.redhat.com/browse/COST-3953)

## Description

This change will allow us to resummarize by provider type. 

Due to the potential hazards of burning down our pipeline, I have add a redis cache that lasts an hour. That will prevent us from triggering the task more than once. 

## Testing

1. To test this PR you will need to start by setting this environment variable to false:
```
(koku) cmyers@cmyers1-mac koku % printenv | grep CACHED
CACHED_VIEWS_DISABLED=False
```

2. load test customer data

3. Resummary Endpoints:
```http://127.0.0.1:5042/api/cost-management/v1/report_data/?provider_type=AWS&start_date=2023-08-01&provider_uuid=*```
* Check the logs and see that *only* the `AWS` sources are kicked off for resummary:
```koku-koku-worker-1  | [2023-08-18 16:17:22,913] INFO f714560a-b0e6-4e78-82b6-978aaf4cc26e 2611 Gathering data for for
koku-koku-worker-1  |  schema_name: org1234567
koku-koku-worker-1  |  provider: AWS-local
koku-koku-worker-1  |  account (provider uuid): 116e92cb-e134-4334-89ed-8217c0b61767
koku-koku-worker-1  | [2023-08-18 16:17:22,916] INFO f714560a-b0e6-4e78-82b6-978aaf4cc26e 2611 Gathering data for for
koku-koku-worker-1  |  schema_name: org1234567
koku-koku-worker-1  |  provider: AWS-local
koku-koku-worker-1  |  account (provider uuid): 4d203980-410e-47bb-a2ae-84081d0f0690
```
4. Try to hit the same endpoint again:
```
http://127.0.0.1:5042/api/cost-management/v1/report_data/?provider_type=AWS&start_date=2023-08-01&provider_uuid=*
```
Notice you will get a return similar to:
```
provider_type (AWS) still disabled for 57.53333333333333 minutes
provider_type (AWS) still disabled for 35.1 minutes
```
5. remove redis-cache
```make delete-redis-cache```
6. Go into the database:
```
DELETE FROM reporting_aws_cost_summary_p WHERE usage_start >= '2023-08-01';
DELETE 36
```
7. Now resummarize a specific time range:
```
http://127.0.0.1:5042/api/cost-management/v1/report_data/?end_date=2023-08-14&provider_type=AWS&start_date=2023-08-06&provider_uuid=*
```
8. Check the database for those times:
```
postgres=# select distinct(usage_start) from reporting_aws_cost_summary_p WHERE usage_start >= '2023-08-01' order by usage_start;
 usage_start 
-------------
 2023-08-06
 2023-08-07
 2023-08-08
 2023-08-09
 2023-08-10
 2023-08-11
 2023-08-12
 2023-08-13
 2023-08-14
 ```

## Notes

...
